include/zfs_namecheck.h

```diff
@@ -43,6 +43,8 @@ typedef enum {
 	NAME_ERR_RESERVED,		/* entire name is reserved */
 	NAME_ERR_DISKLIKE,		/* reserved disk name (c[0-9].*) */
 	NAME_ERR_TOOLONG,		/* name is too long */
+	NAME_ERR_SELF_MOUNT,	/* pool mount is not allowed */
```

Comment:
in the comment, pools cannot be mounted. I think you meant "filesystem"
Perhaps use
```
/* filesystem mount onto self is not allowed */
/* filesystem mount onto parent is not allowed */
```

Reply:
I will replace with those /* self path reference in filesystem is not allowed */
/* parent path reference in filesystem is not allowed */